quickwit/quickwit-search/src/collector.rs

```diff
@@ -593,6 +734,24 @@ impl QuickwitSegmentCollector {
     }
 }
 
+/// This is to circumvent lifetime issues and is pretty terrible.
+///
+/// We either use `filtered_docs` buffer (computed in `compute_filtered_block`) or return the passed
+/// docs.
+#[inline]
+fn get_final_docs<'a>(
```

Comment:
Any better solution for this, it is really quite terrible.

Reply:
I think it can work for instance if we split the SegmentCollector in two...

The filtered_docs and the rest of the member.

Reply:
I replaced it by removing `self` from the top_k collection method and passing the parameters instead